dvc/remote/local/slow_link_detection.py

```diff
@@ -0,0 +1,60 @@
+import logging
+import time
+
+from dvc.config import Config
+from dvc.utils import boxify
+
+logger = logging.getLogger(__name__)
+
+
+class SlowLinkDetectorDecorator(object):
+    LINKING_TIME_SECONDS = 20.0
+    SHOULD_DISPLAY = True
+
+    def __init__(self, method):
+        self.method = method
+
+    def __call__(self, remote_local, *args, **kwargs):
+        start = time.time()
+        result = self.method(remote_local, *args, **kwargs)
+        execution_time_seconds = time.time() - start
+
+        if (
+            execution_time_seconds >= self.LINKING_TIME_SECONDS
+            and self.SHOULD_DISPLAY
+        ):
+            msg = (
+                "\n"
+                + boxify(
+                    "You can cut cache linking time considerably if you "
+                    "will use different link type in your local cache. "
+                    "Check:\n https://dvc.org/doc/commands-reference/"
+                    "config#cache\n for more information.\n To disable "
+                    "this message, run 'dvc config "
+                    "cache.slow_link_warning False'",
+                    border_color="blue",
+                )
+                + "\n"
+            )
+            logger.info(msg)
+            self.SHOULD_DISPLAY = False
```

Comment:
All caps is usually used for constants, so if you wish to modify it you should probably not use all caps, since it is no longer a constant :)